include/engine/datafacade/shared_datafacade.hpp

```diff
@@ -534,22 +535,56 @@ class SharedDataFacade final : public BaseDataFacade
     }
 
     virtual void
-    GetUncompressedWeights(const EdgeID id,
-                           std::vector<EdgeWeight> &result_weights) const override final
+    GetUncompressedReverseGeometry(const EdgeID id,
+                                   std::vector<NodeID> &result_nodes) const override final
     {
-        const unsigned begin = m_geometry_indices.at(id);
+        const signed begin = m_geometry_indices.at(id) - 1;
+        const signed end = m_geometry_indices.at(id + 1) - 2;
+
+        result_nodes.clear();
```

Comment:
@MoKob I think the one thing preventing us from switching to a shared templated function is that reverse iterators don't seem to work on the shared memory `m_geometry_list` or `m_datasource_list`. I'm not sure if this is an easy fix or not.